### PR TITLE
[MRG] Update pybv to write non-voltage channels

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #664 

Once pybv 0.5 is released, the setup.py, requirements.txt file and README can be updated.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/tests/test_write.py [hunk 1]
```diff
@@ -2196,3 +2196,50 @@ def test_sidecar_encoding(_bids_validate):
...
+
+    # the BIDS path for test datasets to get written to
+    bids_path = _bids_path.copy().update(root=bids_root, datatype='eeg')
+
+    raw = reader(raw_fname)
+    kwargs = dict(raw=raw, bids_path=bids_path, overwrite=True)
+
+    # alter some channels manually; in NK and Persyst, this will cause
+    # channel to not have units
+    raw.set_channel_types({raw.info['ch_names'][0]: 'stim'})
```
Comment (adam2392) [inline mne_bids/tests/test_write.py:2221]: I'm trying to use these lines to alter the "units" inside the raw file. However, in the EDF file, when I do this, things still keep them as µV, so might be something to do w/ the raw file itself? 

Otw, the rest of the tests work to validate that other units are written.

**Comment by adam2392** [comment]
Current CI fails until pybv0.5 is released. 

@sappelhoff wdyt? This works locally by pip installing the latest master branch of pybv.

**Comment by sappelhoff** [comment]
I'll get to a release within the next three days 🤓

**Comment by adam2392** [comment]
Happy New Years ;)

**Comment by sappelhoff** [comment]
> Happy New Years ;)

thanks :) for you too!

here we go: https://pypi.org/project/pybv/0.5.0/

**Comment by adam2392** [comment]
This is good to go from my end. The tests are used to manually modify data to become a non-voltage channel.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -41,7 +41,7 @@ API changes
 Requirements
 ^^^^^^^^^^^^
 
-- xxx
+- Updated requirements for ``pybv`` to 0.5 allowing BrainVision written files to contain all types of non-voltage channels, by `Adam Li`_ (:gh:`670`)
```
Comment (sappelhoff) [inline doc/whats_new.rst:44]:
```suggestion
- For writing BrainVision files, ``pybv`` version 0.5 is now required to allow writing of non-voltage channels, by `Adam Li`_ (:gh:`670`)
```

File: mne_bids/tests/test_write.py [hunk 3]
```diff
@@ -2196,3 +2196,50 @@ def test_sidecar_encoding(_bids_validate):
...
+
+    if dir_name == 'EDF':
+        # converting this channel to stim won't change the units
+        return
```
Comment (sappelhoff) [inline mne_bids/tests/test_write.py:2225]: you mention in https://github.com/mne-tools/mne-bids/pull/670/files#r550362399 why this is not working.

I suggest you either:

1. fix it, or
2. remove it

but having it in the code like this is bound to raise an eyebrow in the future (when nobody remembers this brief conversation we are having).

File: mne_bids/write.py [hunk 4]
```diff
@@ -766,19 +765,14 @@ def _write_raw_brainvision(raw, bids_fname, events):
         meas_date = _stamp_to_dt(meas_date)
 
     # pybv currently only supports channels in Volts, except fNIRS
```
Comment (sappelhoff) [inline mne_bids/write.py:767]:
```suggestion
    # pybv needs to know the units of the data for appropriate scaling
```

File: mne_bids/write.py [hunk 5]
```diff
@@ -766,19 +765,14 @@ def _write_raw_brainvision(raw, bids_fname, events):
...
-    if len(ch_idxs) != len(raw.ch_names):
-        non_volt_chs = set(raw.ch_names) - set(np.array(raw.ch_names)[ch_idxs])
-        msg = ('Conversion to BrainVision format needed to be stopped, '
-               'because your raw data contains channel types that are '
-               f'not represented in Volts: "{non_volt_chs}"'
-               '\n\nUntil BrainVision format conversion is improved, you '
-               'must drop these channels from your raw data before using '
-               'mne-bids. Please contact the mne-bids team about this.')
-        raise RuntimeError(msg)
+    # get voltage units as micro-volts and everything else as is
```
Comment (sappelhoff) [inline mne_bids/write.py:768]:
```suggestion
    # get voltage units as micro-volts and all other units "as is"
```

**Review by sappelhoff — COMMENTED** [review]
LGTM except for some minor issues

**Review by sappelhoff — APPROVED** [review]

**Review by hoechenberger — APPROVED** [review]

**Comment by sappelhoff** [comment]
Thanks @adam2392

**Pull request merged**